docs/dictionary/function/processor.lcdoc

```diff
@@ -23,17 +23,15 @@ if the processor is "ARM" then ...
 
 Returns (enum):
 
--   x86: Windows systems (any architecture) and Mac OS X and Linux
-    systems with a 32 bit processor.
--   x86_64: Mac and Linux systems running a 64 bit processor
--   i386: iOS Simulator.
--   ARM: iOS and Android Devices and the Android simulator
-
+-   x86: 32 or 64 bit Windows systems and 32 bit macOS or Linux systems.
```

Comment:
the last case sounds like ample justification. The processor always returned 32-bit on Mac - regardless of processor; as it did on Linux; both before we added 64-bit builds of both. Therefore the natural extension was to return 64-bit if the engine was compiled for 64-bit. So the docs need to reflect that.

The last use-case above is ample justification; as is the use to find the build folder which is arch dependent (on Linux and win as they don't do universal builds).

Indeed it is also justification for the use in check for updates where on linux if you are running a 64-but engine, you probably want a 64-bit update.

So I don't think there is anything to see here - the docs just need to explain that it returns the arch of the processor the engine is built for.

I struggle to find a use-case for knowing the actual processor - since it isn't relevant to running code (as generally 64-bit and 32-but things do not interoperate except at process boundaries, if at all).